dask/array/chunk.py

```diff
@@ -169,54 +179,91 @@ def trim(x, axes=None):
     return x[tuple(slice(ax, -ax if ax else None) for ax in axes)]
 
 
-try:
-    from numpy import broadcast_to
-except ImportError:  # pragma: no cover
-    broadcast_to = npcompat.broadcast_to
+def takeslice(a, indices, axis):
+    """Like `numpy.take`, but allows for indices to be a slice
+    """
+    assert 0 <= axis < a.ndim
+    return a[tuple(indices if i == axis else slice(None)
+                   for i in range(a.ndim))]
 
 
 def topk(a, k, axis, keepdims):
-    """Kernel of topk and argtopk.
+    """Chunk and combine kernel of topk.
     Extract the k largest elements from a on the given axis.
     If k is negative, extract the -k smallest elements instead.
     Note that, unlike in the parent function, the returned elements
     are not sorted internally.
     """
+    assert keepdims is True
     axis = axis[0]
     if abs(k) >= a.shape[axis]:
         return a
+
     a = np.partition(a, -k, axis=axis)
-    # return a[-k:] if k>0 else a[:-k], on arbitrary axis
-    return a[[
-        (slice(-k, None) if k > 0 else slice(None, -k))
-        if i == axis else slice(None)
-        for i in range(a.ndim)
-    ]]
+    k_slice = slice(-k, None) if k > 0 else slice(-k)
+    return takeslice(a, k_slice, axis=axis)
 
 
-def topk_postprocess(a, k, axis):
-    """Kernel of topk and argtopk.
-    Post-processes the output of topk, sorting the results internally.
+def topk_aggregate(a, k, axis, keepdims):
+    """Final aggregation kernel of topk.
+    Invoke topk one final time and then sort the results internally.
```

Comment:
Nitpick, include a space or endline after the first `"""` and separate the header line by an empty line.  See http://numpydoc.readthedocs.io/en/latest/ or http://dask.pydata.org/en/latest/develop.html#docstrings